### PR TITLE
Implementation of Explorer Component

### PR DESCRIPTION
**Pull request by crhallberg**
Making a PR early to open discussion about the progress and path of this component.

TODO:
 - [x] Starting with a manifest that is not the root
 - [x] Styling
 - [x] Switch away from Font Awesome, likely to CSS and small images for icons
 - [x] Reduce __jsonld
   - [x] getLabel() and getProperty()

After merge:
 - [ ] Add preview thumbnails option
   - [ ] If thumbnails are on, how do we distinguish collections from records?
 - [ ] Handle records with multiple parents (a multivalued `within` property)
 - [ ] Adapt to UV code standards

### CODE REVIEW

File: src/ExplorerComponent.ts [hunk 1]
```diff
@@ -1,43 +1,172 @@
...
+                let that = this;
+                node.data.load().then(function (collection: Manifesto.Collection): void {
+                    console.log(collection);
+                    node.nodes = collection.members.map(function (op: Manifesto.IIIFResource): Manifesto.TreeNode {
+                        // TODO: OFFICIAL CONVERSION MUST EXIST
+                        let data: any = op;
+                        data.type = op.__jsonld['@type'].toLowerCase();
+                        let treeNode: Manifesto.TreeNode = new Manifesto.TreeNode(op.__jsonld.label, data);
+                        treeNode.parentNode = node;
+                        return treeNode;
```
Comment (crhallberg) [inline src/ExplorerComponent.ts]: I'm concerned about this part of the code, I have yet to find a better, pre-existing way to convert IIIFResources to TreeNodes. It must exist.

File: typings/manifold.bundle.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// manifesto.js v1.0.0 https://github.com/viewdir/manifesto
+// manifesto.js v2.0.1 https://github.com/viewdir/manifesto
```
Comment (crhallberg) [inline typings/manifold.bundle.d.ts:1]: Updated to fix errors. Is there more updated code we can merge in?

Comment (edsilv) [reply typings/manifold.bundle.d.ts:1]: Looks like you're on the latest code. On the last Software Developer's call we discussed whether TreeNode should be moved to its own separate repo. My feeling is that manifesto shouldn't really be generating the trees - this is an "opinionated" view over the ranges graph for a viewer. Manifold should generate the tree instead. TreeNode and perhaps specialised File/Folder classes for this component could be moved to a generic repo that has no specific relation to IIIF concepts. Does this sound reasonable to you?

Comment (crhallberg) [reply typings/manifold.bundle.d.ts:1]: I'd support that decision. Actually in my latest pushes I removed TreeNode usage for straight Manifest uses.

**Comment by crhallberg** [comment]
Realized that this addresses [issue #329](https://github.com/UniversalViewer/universalviewer/issues/329) from the UV 2.0 project board.

**Comment by crhallberg** [comment]
A big concern right now is how much styling to put into this, considering the UV has such a strong theme of its own.

**Pull request merged**